rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -211,22 +221,19 @@ private class InMemHotStore[F[_]: Concurrent, C, P, A, K](
   def getJoins(channel: C): F[Seq[Seq[C]]] =
     for {
       fromHistoryStore <- getJoinsFromHistoryStore(channel)
-      result <- hotStoreState.modify[Seq[Seq[C]]](state => {
-                 state.joins.get(channel) match {
-                   // update with what is in historyStore and return the same
-                   case None =>
-                     (
-                       state.copy(
-                         joins = state.joins.updated(channel, fromHistoryStore)
-                       ),
-                       state.installedJoins.getOrElse(channel, Seq.empty) ++: fromHistoryStore
-                     )
-                   // just return what is in hot store already
-                   case Some(joins) =>
-                     (state, state.installedJoins.getOrElse(channel, Seq.empty) ++: joins)
-                 }
-               })
-
+      result <- hotStoreState.modify[Seq[Seq[C]]] { state =>
+                 // update with what is in historyStore and return the same if empty...
+                 val historyStoreIfEmpty = (
+                   state.copy(joins = state.joins.updated(channel, fromHistoryStore)),
+                   state.installedJoins.getOrElse(channel, Seq.empty) ++: fromHistoryStore
+                 )
+                 // ...or just return what is in hot store already
+                 state.joins
+                   .get(channel)
+                   .fold(historyStoreIfEmpty)(
+                     joins => (state, state.installedJoins.getOrElse(channel, Seq.empty) ++: joins)
+                   )
+               }
```

Comment:
Try to rewrite this by using `state.joins.get(channel).map(...)`. We don't want to execute any operation on a _state_ if not necessary.
We can define `def historyStoreIfEmpty` but I'd say it makes things less clear.

Reply:
Done